src/main/java/com/google/sps/utils/data/DataCollection.java

```diff
@@ -0,0 +1,235 @@
+   * @param species: non-null species that isn't already stored in Datastore
+   */
+  public static void addApiInfo(Species species) {
+    // Add API-side fields if available
+    try {
+      String apiJSON = SpeciesAPIRetrieval.getJSON(API_URL, species.getBinomialName());
+      Map apiMap = SpeciesAPIRetrieval.convertJSONToMap(apiJSON);
+      SpeciesAPIRetrieval.addAPISpeciesInfo(species, apiMap);
+    } catch (Exception e) {
+      System.out.println("Exception occurred.");
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Store species in Datastore
+   * @param species: species to store
+   */
+  public static void addSpeciesToDatastore(Species species) {
+    Key key = keyFactory.newKey(species.getBinomialName());
+    Entity oldEntity = datastore.get(key);
+    
+    if (oldEntity == null) {
+      Entity speciesEntity = Entity.newBuilder(key)
+        .set("common_name", species.getCommonName())
+        .set("binomial_name", species.getBinomialName())
+        .set("trend", species.getTrend().name())
+        .set("status", species.getStatus())
+        .set("population", species.getPopulation())
+        .set("kingdom", species.getTaxonomy().getAnimalKingdom())
+        .set("phylum", species.getTaxonomy().getAnimalPhylum())
+        .set("class", species.getTaxonomy().getAnimalClass())
+        .set("order", species.getTaxonomy().getAnimalOrder())
+        .set("family", species.getTaxonomy().getAnimalFamily())
+        .set("genus", species.getTaxonomy().getAnimalGenus())
+        .set("image_link", species.getImageLink())
+        .set("wikipedia_notes", species.getWikipediaNotes())
+        .set("citation_link", species.getCitationLink())
+        .build();
+      
+      datastore.put(speciesEntity);
+    //   System.out.println(species.getBinomialName() + " was added to Datastore.");
+    // } else {
+    //   System.out.println(species.getBinomialName() + " was already in Datastore.");
+    }
+  }
+
+  // ------------------------------  SCRAPING HELPER FUNCTIONS  ------------------------------ //
+  private static String scrapeStatus(String statusString) {
+    String status = statusString.replaceAll("Domesticated", "DO");
+    status = removeBrackets(status);
+    return status;
+  }
+
+  private static String scrapePopulation(String populationString) {
+    String pop = removeBrackets(populationString);
+    pop = pop.replaceAll("[^0-9.–-]", "");
+    return pop;
+  }
+
+  private static PopulationTrend scrapeTrend(Element trendImg) {
+    String trend;
+    if (trendImg != null) {
+      trend = trendImg.attr("alt");
+    } else {
+      trend = "Unknown";
+    }
+    return convertToPopulationTrendEnum(trend);
+  }
+
+  private static String scrapeImageLink(Element image) {
+    String imageLink = "";
+    if (image != null) {
+      imageLink = image.absUrl("src");
+    }
+    return imageLink;
+  }
+
+  public static PopulationTrend convertToPopulationTrendEnum(String trendString) {
+    switch(trendString) {
```

Comment:
Since Wikipedia gives the trend as "Increase", "Decrease", etc., would it be alright to change the enum values to match those (since the naming convention for enums is all caps)?

Reply:
@arianazhu, we could probably use .toUpperCase() to get it in all caps. I think we need to change the enum from INCREASING to INCREASE though for this to work.

Reply:
Oops, didn't think of that! Good idea